### DISCUSSION TITLE
Update existing data in Prometheus database

### DISCUSSION THREAD

**Discussion by benisai** [prometheus/prometheus#11327] [Q&A]
## Proposal
**Use case. Why is this important?**
My servers/networking device's are importing nat traffic 
data  into Prometheus (src and dest IP addresses). I'd like to convert this data into fqdn/hostnames.  I would like to know if there is a method of doing so? Is there a way to interact with the database? Similar to SQL? 

I don't want to lose my existing data, so I'd like some help or a script that can update these IP addresses to hostnames.

**Comment by benisai** [comment]
Anyway to update data that already exists in Prometheus? Is there a shell script or something that can connect to the Prometheus database? I really need help

**Comment by LeviHarrison** [comment]
The Prometheus TSDB is for the most part immutable, so there isn't a way to change existing data.
